src/solver/batchnorm/forward_inference_ck.cpp

```diff
@@ -126,52 +126,60 @@ template <typename XDataType,
           typename ScaleDataType,
           typename BiasDataType,
           typename MeanVarDataType>
-static void RunCKSolution(const Handle& handle,
-                          const AnyInvokeParams& primitive_parameters,
-                          const miopen::batchnorm::ProblemDescription& problem)
+ConvSolution InvokerFactoryMakerNHWC(const miopen::batchnorm::ProblemDescription& bn_problem)
 {
-    const auto& args = CKArgsBNormFwd{problem};
-
-    using DeviceOp = ck::tensor_operation::device::DeviceElementwise<
-        ck::Tuple<XDataType, MeanVarDataType, MeanVarDataType, ScaleDataType, BiasDataType>,
-        ck::Tuple<YDataType>,
-        Normalize,
-        Rank>;
-    const auto bn_fwd_ptrs = ck::tensor_operation::device::instance::DeviceOperationInstanceFactory<
-        DeviceOp>::GetInstances();
+    ConvSolution result;
+    result.invoker_factory = [args         = CKArgsBNormFwd{bn_problem},
+                              kernel_index = CheckCKApplicability<XDataType,
+                                                                  YDataType,
+                                                                  AccDataType,
+                                                                  ScaleDataType,
+                                                                  BiasDataType,
+                                                                  MeanVarDataType>(bn_problem)](
+                                 const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [&](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
```

Comment:
```suggestion
        return [=, args = std::move(args)](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
```

Reply:
@DrizztDoUrden , thanks for the comments. Code updated